wgpu-core/src/lib.rs

```diff
@@ -136,6 +136,46 @@ fn loom() {
     });
 }
 
+/// Reference count object that tracks multiple references.
+#[derive(Debug)]
+struct MultiRefCount(ptr::NonNull<AtomicUsize>);
+
+unsafe impl Send for MultiRefCount {}
+unsafe impl Sync for MultiRefCount {}
+
+impl MultiRefCount {
+    fn new() -> Self {
+        let bx = Box::new(AtomicUsize::new(1));
+        MultiRefCount(unsafe { ptr::NonNull::new_unchecked(Box::into_raw(bx)) })
+    }
+
+    fn inc(&self) {
+        unsafe { self.0.as_ref() }.fetch_add(1, Ordering::Relaxed);
```

Comment:
Is Relaxed + Release the correct ordering for atomic reference counting? (Not saying it's wrong, just verifying this is intentional and correct). I remember the pair being a weird combination like this.

Reply:
That was iffy a bit. We have #150 for general audit of this, but I added a commit that makes these more close to what we need. Please have a look!